students/scotchwsplenda/Lesson_6/MailRoom_4.py

```diff
@@ -0,0 +1,144 @@
+
+from operator import itemgetter
+import sys
+
+
+# donors
+donators = {"Gordian": [30.0, 45.0], "Tiberius": [60.0],
+            "Maximus": [65.0, 12.0], "Tacitus": [33.0, 22.0, 25.00],
+            "Commodus": [43.0, 11.0]}
+
+
+def mainy():
+    print('''
+  Welcome to the MAILROOM main menu
+  Please choose from below options:
+  1 - Initiate Grovel Sequence
+  2 - Data Metrics 3000
+  3 - Quit
+  4 - Send thank you notes to all donors
+              ''')
+    while True:
+        try:
+            choice = int(input('Main Indicate your choice: '))
+            swit_dic = {1: send_note, 2: data_metrics,
+                        3: exit_program, 4: mass_mail}
+            swit_dic[choice]()
+        except KeyError:
+            print('You have entered a non-choice'
+                  ' , please get your shit together')
+        except ValueError:
+            print('You have entered a non-number'
+                  ' , please get your shit together')
+            continue
+
+
+# Option 1: Content of Thank you note
+def send_note():
+    print('''
+Please select from the below Thank You Note options:
+1 - Print list of extant donors
+2 - Enter donor name and donation
+3 - Exit Program
+    ''')
+    while True:
+        try:
+            respondy = int(input("send_note: Indicate your choice:"))
+            if respondy == 1:
+                print_donors()
+                continue
+            if respondy == 2:
+                try:
+                    donor_inp = input("Input donor: ")
+                    if donor_inp in donators.keys():
+                        don_amount = input("That's a known donor,"
+                                           " input donation amount: ")
+                    else:
+                        don_amount = input(f"{donor_inp}'s an unknown donor,"
+                                           " input donation amount: ")
+                    don_amount = float(don_amount)
+                    update_dons(donor_inp, don_amount)
+                    send_thanks(donor_inp, don_amount)
+                    continue
+                except ValueError:
+                    print('You have entered a non-number'
+                          ' , for the donation amount start over')
+            if respondy == 3:
+                mainy()
+            if respondy > 3:
+                print('You have entered a non-choice'
+                      ' , please get your shit together')
+                continue
+            if respondy < 1:
+                print('You have entered a non-choice'
+                      ' , please get your shit together')
+                continue
+        except ValueError:
+            print('You have entered a non-number'
+                  ' , please get your shit together')
+            continue
```

Comment:
you don't need any of the `continue` statements above. `continue` is used if there is no other code lines or you need to skip some code underneath the continue